packages/advanced-logic/specs/payment-network-erc777-stream-0.1.0.md

```diff
@@ -0,0 +1,445 @@
+
+There are 2 kinds of request extensions in order to accept ERC777:
+
+- The first request of a series is very similar to [payment-network-erc20-fee-proxy](./payment-network-erc20-fee-proxy-contract-0.1.0.md), it defines the salt, paymentAddress and requestId to compute the `paymentReference` used for the whole series.
+- Other requests must define a `previousRequestId` and cannot redefine any of the payment properties (`paymentAddress`, `feeAddress` or `salt`).
+
+This version 0.1.0 does not allow refunds to be streamed. Refunds have to be made according to [payment-network-erc20-fee-proxy](./payment-network-erc20-fee-proxy-contract-0.1.0.md), using the fact that ERC777 tokens implement the ERC20 interface.
+
+?? remaining questions ??
+
+- impact on reputation
+- flow rate < invoicing rate ?
+
+As for [ERC20 Fee Proxy Contract](./payment-network-erc20-proxy-contract-0.1.0.md) requests, the request issuer can also declare payments manually. Fees shall not be paid for declarative payments.
+
+The SuperApp contract does the ERC777 token stream on behalf of the user. The contract ensures a link between an ERC777 SuperFluid stream and a request through a `paymentReference` stored in the userdata of the incoming stream. This `paymentReference` consists of the last 8 bytes of a salted hash of the requestId: `last8Bytes(hash(lowercase(requestId + salt + address)))`:
+
+As a payment network, this extension describes how to deduce a payment `balance` for the request. (see [Interpretation](#Interpretation))
+
+## Payment SuperApp Contract
+
+The contract contains one function called `streamFromWithReferenceAndFee` which takes 6 arguments:
+
+- `tokenAddress` is the address of the ERC777 contract
+- `to` is the destination address for the tokens
+- `amount` is the amount of tokens to stream to the destination address
+- `paymentReference` is the reference data used to track the stream (see `paymentReference`)
+- `feeAmount` is the amount of tokens to stream to the fee destination address
+- `feeAddress` is the destination address for the fee
+
+The `StreamWithReferenceAndFee` event is emitted when the tokens are streamed. This event contains the same 6 arguments as the `streamFromWithReferenceAndFee` function.
+
+[See smart contract source](https://github.com/RequestNetwork/requestNetwork/blob/master/packages/smart-contracts/src/contracts/ERC777FeeSuperApp.sol)
+
+| Network | Superhost Address                          |
+| ------- | ------------------------------------------ |
+| Private | FIXME                                      |
+| matic   | 0x1EB3FAA360bF1f093F5A18d21f21f13D769d044A |
+| xDai    | 0x8e8F05f1aD16D20e66Bd0922b510332104ddAc7B |
+
+## Manual payment declaration
+
+Manual payment declarations follow the same specifications as [payment-network-erc20-fee-proxy](./payment-network-erc20-fee-proxy-contract-0.1.0.md).
+
+## Properties
```

Comment:
CR hint: this is quite original, challenges welcome

Reply:
it seems the easiest way to connect series of requests.